### PR TITLE
Add support for ignoreMigrationPatterns config in Flyway

### LINKED ISSUE #27099: Support `ignoreMigrationPatterns` config in Flyway
### Description

As requested in https://github.com/quarkusio/quarkus/pull/27096#discussion_r936069360, Flyway supports configuring certain patterns to ignore migrations during validate and repair.
It would be nice to be able to specify that in the Flyway Config. 

More info: https://flywaydb.org/documentation/configuration/parameters/ignoreMigrationPatterns

### Implementation ideas

Include a `quarkus.flyway.ignore-migration-patterns` in the [FlywayDataSourceRuntimeConfig](https://github.com/quarkusio/quarkus/blob/main/extensions/flyway/runtime/src/main/java/io/quarkus/flyway/runtime/FlywayDataSourceRuntimeConfig.java) and create a test for it.

See [`c31af2a` (#27096)](https://github.com/quarkusio/quarkus/pull/27096/commits/c31af2a05ffcec623762812a6176fafc62434d7c) for an example

### PR DESCRIPTION
**Pull request by tmihalac**
Added `quarkus.flyway.ignore-migration-patterns` parameter to quarkus config.

Resolves https://github.com/quarkusio/quarkus/issues/27099

### CODE REVIEW

File: extensions/flyway/runtime/src/main/java/io/quarkus/flyway/runtime/FlywayDataSourceRuntimeConfig.java [hunk 1]
```diff
@@ -187,4 +187,11 @@ public static final FlywayDataSourceRuntimeConfig defaultConfig() {
      */
     @ConfigItem
     public boolean validateMigrationNaming;
+
+    /**
+     * Ignore migrations during validate and repair according to a given list of patterns.
+     * Patterns are comma seperated.
+     */
+    @ConfigItem
+    public Optional<String> ignoreMigrationPatterns = Optional.empty();
```
Comment (gastaldi) [inline extensions/flyway/runtime/src/main/java/io/quarkus/flyway/runtime/FlywayDataSourceRuntimeConfig.java:196]: I believe you can use `List<String>` or `String[]` here (wrapped in an `Optional`)

File: extensions/flyway/runtime/src/main/java/io/quarkus/flyway/runtime/FlywayDataSourceRuntimeConfig.java [hunk 2]
```diff
@@ -187,4 +187,11 @@ public static final FlywayDataSourceRuntimeConfig defaultConfig() {
      */
     @ConfigItem
     public boolean validateMigrationNaming;
+
+    /**
+     * Ignore migrations during validate and repair according to a given list of patterns.
+     * Patterns are comma seperated.
```
Comment (gastaldi) [inline extensions/flyway/runtime/src/main/java/io/quarkus/flyway/runtime/FlywayDataSourceRuntimeConfig.java:193]:
```suggestion
     * Patterns are comma separated.
```

**Review by gastaldi — CHANGES_REQUESTED** [review]

File: extensions/container-image/container-image-jib/deployment/src/main/java/io/quarkus/container/image/jib/deployment/JibConfig.java [hunk 3]
```diff
@@ -161,16 +161,16 @@ public class JibConfig {
...
      *
      * If more than one platform is configured, it is important to note that the base image has to be a Docker manifest or an
-     * OCI image index containing a version of each chosen platform
+     * OCI image index containing a version of each chosen platform.
      *
-     * It doesn't work with native images, as cross-compilation is not supported
+     * The feature does not work with native images, as cross-compilation is not supported.
      *
-     * Target Platform is a incubating feature of Jib. See <a href=
+     * This configuration is based on an incubating feature of Jib. See <a href=
```
Comment (gastaldi) [inline extensions/container-image/container-image-jib/deployment/src/main/java/io/quarkus/container/image/jib/deployment/JibConfig.java:171]: Although it is nice, this change is not related to the PR proposal, it should be in a separate commit

File: extensions/flyway/runtime/src/main/java/io/quarkus/flyway/runtime/FlywayDataSourceRuntimeConfig.java [hunk 4]
```diff
@@ -187,4 +187,11 @@ public static final FlywayDataSourceRuntimeConfig defaultConfig() {
      */
     @ConfigItem
     public boolean validateMigrationNaming;
+
+    /**
+     * Ignore migrations during validate and repair according to a given list of patterns.
+     * Patterns are comma separated.
+     */
+    @ConfigItem
+    public Optional<String[]> ignoreMigrationPatterns = Optional.empty();
```
Comment (gastaldi) [inline extensions/flyway/runtime/src/main/java/io/quarkus/flyway/runtime/FlywayDataSourceRuntimeConfig.java:196]: Not that important, but there is no need to explicitly initialize the attribute here

Comment (tmihalac) [reply extensions/flyway/runtime/src/main/java/io/quarkus/flyway/runtime/FlywayDataSourceRuntimeConfig.java]: If I don't initialize it here I need to check everywhere I do `flywayRuntimeConfig.ignoreMigrationPatterns.isPresent()` for nullity before and I think it defeats the purpose of using `Optional`

Comment (gastaldi) [reply extensions/flyway/runtime/src/main/java/io/quarkus/flyway/runtime/FlywayDataSourceRuntimeConfig.java:196]: Only if you create the config manually (when in unit tests for example). When injecting it this is never null

Comment (tmihalac) [reply extensions/flyway/runtime/src/main/java/io/quarkus/flyway/runtime/FlywayDataSourceRuntimeConfig.java:196]: Is it ok to leave it like this ?

Comment (gastaldi) [reply extensions/flyway/runtime/src/main/java/io/quarkus/flyway/runtime/FlywayDataSourceRuntimeConfig.java:196]: Sure, no problem

File: extensions/flyway/runtime/src/main/java/io/quarkus/flyway/runtime/FlywayDataSourceRuntimeConfig.java [hunk 5]
```diff
@@ -187,4 +187,11 @@ public static final FlywayDataSourceRuntimeConfig defaultConfig() {
      */
     @ConfigItem
     public boolean validateMigrationNaming;
+
+    /**
+     * Ignore migrations during validate and repair according to a given list of patterns.
+     * Patterns are comma separated.
```
Comment (gastaldi) [inline extensions/flyway/runtime/src/main/java/io/quarkus/flyway/runtime/FlywayDataSourceRuntimeConfig.java:193]: I think it is important to mention that when this configuration is set, the `ignoreFutureMigrations` and `ignoreMissingMigrations` settings are ignored

File: extensions/flyway/runtime/src/main/java/io/quarkus/flyway/runtime/FlywayDataSourceRuntimeConfig.java [hunk 6]
```diff
@@ -187,4 +187,12 @@ public static final FlywayDataSourceRuntimeConfig defaultConfig() {
      */
     @ConfigItem
     public boolean validateMigrationNaming;
+
+    /**
+     * Ignore migrations during validate and repair according to a given list of patterns,
```
Comment (gastaldi) [inline extensions/flyway/runtime/src/main/java/io/quarkus/flyway/runtime/FlywayDataSourceRuntimeConfig.java:192]: It isn't totally clear what patterns are these, maybe add a link to https://flywaydb.org/documentation/configuration/parameters/ignoreMigrationPatterns here or include the pattern format from that page here?

File: extensions/flyway/runtime/src/main/java/io/quarkus/flyway/runtime/FlywayDataSourceRuntimeConfig.java [hunk 7]
```diff
@@ -187,4 +187,13 @@ public static final FlywayDataSourceRuntimeConfig defaultConfig() {
      */
     @ConfigItem
     public boolean validateMigrationNaming;
+
+    /**
+     * Ignore migrations during validate and repair according to a given list of patterns
+     * https://flywaydb.org/documentation/configuration/parameters/ignoreMigrationPatterns,
+     * when this configuration is set, the ignoreFutureMigrations and ignoreMissingMigrations settings are ignored
+     * Patterns are comma separated.
```
Comment (gastaldi) [inline extensions/flyway/runtime/src/main/java/io/quarkus/flyway/runtime/FlywayDataSourceRuntimeConfig.java:195]:
```suggestion
     * Ignore migrations during validate and repair according to a given list of patterns (see https://flywaydb.org/documentation/configuration/parameters/ignoreMigrationPatterns for more information).
     * When this configuration is set, the ignoreFutureMigrations and ignoreMissingMigrations settings are ignored. Patterns are comma separated.
```

**Review by gastaldi — APPROVED** [review]
LGTM
